internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -105,20 +110,82 @@ func (p *processRunner) build() (cmd *exec.Cmd, logCloser io.Closer, err error)
 		writer = w
 	}
 
-	cmd.Stdout = writer
-	cmd.Stderr = writer
+	var afterStartFuncs []func()
+
+	if p.opts.StdinFile != "" {
+		stdin, err := os.Open(p.opts.StdinFile)
+		if err != nil {
+			return commandWrapper{}, err
```

Comment:
`w` is not closed on all these exit paths

Reply:
yep, fixed